### PR TITLE
Allow delete of book title and text

### PR DESCRIPTION
**Pull request by paanrama**
For your review, this fix Issue #1743, I only have not decide yet if a default:book_written without title and text must be reverted to a default:book or no (I will wait for your comments).

PD: This is my first time doing a pull request (and using GitHub), so please guide me if I make some mistakes.

### CODE REVIEW

**Comment by paanrama** [comment]
First, thanks for your suggestions!

I make the suggested changes in my computer, tested them, and upload the changes to my minetest fork, also I compare it to official minetest_game master.

Hummm, but I don't understand yet why the changes appear now in this pull request if I don't remember pushing the changes, ... it seems that I must read much more in Github help because there are a lot to learn.

One last question about the change, A written book with its content (title and text) deleted must be converted back to a book (unwritten)? Or is better leave it as a empty written book?

**Comment by srifqi** [comment]
In my opinion, if a book has its content deleted, it becomes "unwritten" (fresh book).

File: mods/default/craftitems.lua [hunk 1]
```diff
@@ -79,12 +79,13 @@ local max_text_size = 10000
 local max_title_size = 80
 local short_title_size = 35
 minetest.register_on_player_receive_fields(function(player, formname, fields)
-	if formname ~= "default:book" then return end
```
Comment (SmallJoker) [inline mods/default/craftitems.lua]: Such a filter is required to avoid side-effects when submitting formspecs which have the same or similar `fields`.

**Comment by paanrama** [comment]
I hope that you will consider Ok this last commit :).

I will be waiting your review. Thanks for your attention.

File: mods/default/craftitems.lua (hunk 1 above)
Comment (paanrama) [reply mods/default/craftitems.lua]: Sorry, I will fix it immediatelly.

**Comment by tenplus1** [comment]
Erasing a book by removing the title won't work unless the person actually owns the book itself, and as we know players give out books containing text a lot.  Why not add a recipe to take a written book, add say a bucket or water (or whatever works better) and return an empty book.

**Comment by paanrama** [comment]
Sorry for the delay (I am ill). 

I was thinking about the recipe method of erase a book, and it seems that will remove the owner of the book concept.

I think that is a too big change for this moment, and need more discussion.

**Comment by paramat** [comment]
I'll close for the moment then, can re-open on progress.

**Pull request closed without merging**